src/lib/webpack/webpack-base-config.js

```diff
@@ -97,6 +100,37 @@ export default (env) => {
 			}
 		}),
 
+		// TypeScript
+		customConfig({
+			module: {
+				loaders: [
+					{
+						enforce: 'pre',
+						test: /\.tsx?$/,
+						use: [
+							{
+								loader: resolve(__dirname, './npm-install-loader'),
+								options: {
+									modules: ['typescript', 'awesome-typescript-loader'],
+									save: true
+								}
+							},
+							{
+								loader: 'awesome-typescript-loader',
+								options: {
+									sourceMap: true,
+									useBabel: true,
+									babelOptions: createBabelConfig(env),
+									useCache: true,
+									configFileName: tsconfig,
```

Comment:
@developit @marvinhagemeister should I add above to the `preact.config.js` recipes and abandon this PR ?